packages/ERTP/core/mint.js

```diff
@@ -121,6 +107,47 @@ allegedName must be truthy: ${allegedName}`;
     return payment;
   }
 
+  // depositInto always deposits the entire payment units
+  function depositInto(purse, payment) {
+    const oldPurseUnits = purseKeeper.getUnits(purse);
+    const paymentUnits = paymentKeeper.getUnits(payment);
+    // Also checks that the union is representable
+    const newPurseUnits = unitOps.with(oldPurseUnits, paymentUnits);
+
+    // ///////////////// commit point //////////////////
+    // All queries above passed with no side effects.
+    // During side effects below, any early exits should be made into
+    // fatal turn aborts.
+    paymentKeeper.remove(payment);
+    purseKeeper.updateUnits(purse, newPurseUnits);
+
+    return paymentUnits;
+  }
+
+  const makeMintContext = harden({
+    takePayment,
+    takePaymentAndKill,
+    depositInto,
+    get assay() {
```

Comment:
Hey David, is there a benefit to doing it this way? 

If it's meant to prevent setting the values, I think harden covers that for us.

Reply:
The initial idea here is to provide the same context object to all customisation methods

However, because they are some dependencies between one another, some objects may not be built the first times the objects is passed. For instance, while building the `assay` object, the `assay` property cannot be already built. Same goes for `unitOps` which can only exist after the creation of `assay`, etc.

So the trick i'm using is defining a getter and that works for all the cases:
- During `makeMint`:
    - either the property is not supposed to exist (like for `.assay` in `makeAssayTrait`) and the person shouldn't even be trying to reach it, but if they do before initialization, it'll throw (because of `const` temporal dead zone)
    - or the property exists and may be expected (most properties in most `make*Trait` functions) and you get the value you want
- Once the `makeMint` call is successful, all custom methods have access to all context properties

Other options without getters include:
- passing a new object every time
    - but it takes some bookkeeping to know which property is supposed to be available for which method
- passing the same un-hardened object to different methods and add properties as they come to exist
    - but malicious `make*Trait` methods may alter the context object and maybe ruin the process. I'm not clear on how big a risk this is and how big the consequences are; the getter trick lets me skip this analysis altogether

Reply:
That makes sense to me. Thanks for explaining.

Reply:
Me too. This is a good use of getters and temporal dead zones for const variables.